engines/wasmer/wasmer.go

```diff
@@ -95,28 +97,47 @@ type (
 var _ = (wapc.Module)((*Module)(nil))
 var _ = (wapc.Instance)((*Instance)(nil))
 
-var engineInstance = engine{}
+var engineInstance = engine{newRuntime: DefaultRuntime}
 
+// Engine returns a new wapc.Engine which uses the DefaultRuntime.
 func Engine() wapc.Engine {
```

Comment:
Instead of having separate `Engine` and `EngineWithRuntime` functions, I wonder about using the Functional Options pattern here. It would be backward compatible. Is there any reason why we wouldn't use it?

https://golang.cafe/blog/golang-functional-options-pattern.html

Reply:
This can be consistent and backward compatible with functional options, but thinking about it more there might be an advantage to having the signature for `Engine` be the same (passing to higher order functions). Let's keep this as is.